src/olympia/addons/models.py

```diff
@@ -72,10 +84,8 @@ def clean_slug(instance, slug_field='slug'):
         translations = Translation.objects.filter(id=instance.name_id)
```

Comment:
nit: the comment above still says the last resort is the model name, but it's a random slug instead now.